shared/fight.h

```diff
@@ -1,5 +1,6 @@
 #pragma once
-#include "gamedata.h"
+#include "client_gamedata.h"
```

Comment:
if we place the implementation from the .cpp file back into the .h file(this file) we don't need these 2 includes anymore and we can place them into server/client directories